### PR TITLE
fix(migration): correctly extract name from migrations and check for …

### LINKED ISSUE #24: Normalize "name" of migration documents
Currently the name is the full absolute path to the file on disk. This is a problem because the status of the migration depends on where the application is located.
I propose that we only use the file name, and add a check to avoid multiple identical names (in case multiple migration folders are configured)

### PR DESCRIPTION
**Pull request by Xstoudi**
…duplicates.

Should fix #24

### CODE REVIEW

File: commands/MongodbMigrate.ts [hunk 1]
```diff
@@ -23,7 +25,23 @@ export default class MongodbMigrate extends MigrationCommand {
...
+    }));
+
+    const duplicates = new Set(
+      migrations.filter(
+        ({ name }, index) =>
+          migrations.map((migration) => migration.name).indexOf(name) !== index,
+      ),
+    );
+    if (duplicates.size > 0) {
+      throw new Error(`found duplicate migration file names`);
```
Comment (targos) [inline commands/MongodbMigrate.ts:42]: could be nice to print the names

File: commands/MongodbMigrate.ts [hunk 2]
```diff
@@ -62,10 +80,12 @@ export default class MongodbMigrate extends MigrationCommand {
         const dbMigrationNames = migrationDocs.map((m) => m.name);
 
         // Keep migrations that are not yet registered
-        migrationFiles = migrationFiles.filter(
-          (name) => !dbMigrationNames.includes(name),
+        const unregisteredMigrations = migrations.filter(
+          (migration: { name: string; file: string }) =>
+            !dbMigrationNames.includes(basename(migration.name, '.js')),
```
Comment (targos) [inline commands/MongodbMigrate.ts:85]: `migration.name` is already basenamed, no?

File: commands/util/MigrationCommand.ts [hunk 3]
```diff
@@ -57,32 +57,30 @@ export default abstract class MongodbMakeMigration extends BaseCommand {
             }
           }),
       )
-    )
-      .filter((migrationFile) => migrationFile !== null)
-      .flat() as string[];
+    ).filter((migrationFile) => migrationFile !== null) as unknown) as string[])
```
Comment (stropitek) [inline commands/util/MigrationCommand.ts:60]: The parenthesis nesting  makes the code really difficult to read

Maybe use an intermediate variable?

**Review by stropitek — COMMENTED** [review]
SGTM if you address those few comments

File: commands/MongodbMigrate.ts [hunk 4]
```diff
@@ -39,7 +39,11 @@ export default class MongodbMigrate extends MigrationCommand {
       ),
     );
     if (duplicates.size > 0) {
-      throw new Error(`found duplicate migration file names`);
+      throw new Error(
+        `found duplicate migration file names: ${[...duplicates]
+          .map(({ name }) => name)
```
Comment (targos) [inline commands/MongodbMigrate.ts:44]: I didn't realize it was an array of objects. Aren't they all different, even if they some share the same name property?

Comment (Xstoudi) [reply commands/MongodbMigrate.ts:44]: They are different, in some case we could have multiple with the same name.

Comment (targos) [reply commands/MongodbMigrate.ts:44]: A test would be very useful to confirm the implementation. What about moving the normalization and validation to the `getMigrationFiles` function or some other function called by it ?

**Review by targos — APPROVED** [review]

**Pull request merged**